### ISSUE TITLE
Change/feature request - Sorting of Groups in Webui file summary endpoint

### ISSUE THREAD

**Issue by fearnlj01** [ShokoAnime/ShokoServer#1130]
Raising this as something to address in the future on @ElementalCrisis's request.

## What/Why
In the `/WebUI/Series/{seriesID}/FileSummary` endpoint, you can end up with groups coming in containing `Special` type episodes only before that of regular episodes, or mixed episode type groups as per the following screenshot.

I would suspect that it makes more sense for the ordering to be addressed serverside rather than by re-sorting the response, hence raising this on Server & not in the WebUI.

![image](https://github.com/ShokoAnime/ShokoServer/assets/13705865/10fd2b3b-01e7-434c-b623-96c760c43137)

## Resolution
I would expect that the desired sorting could _probably_ be attained by ordering the result groups (as set in [`/Shoko.Server/API/v3/Models/Shoko/WebUI.cs`](https://github.com/ShokoAnime/ShokoServer/blob/99f1917cc365635649d1a64da8cdbb328e334a4a/Shoko.Server/API/v3/Models/Shoko/WebUI.cs#L327)) by the count of included episodes, then the count of included specials, and so on.

As to if this is a good idea or not, I suspect not, but it's 'something' to consider rather than nothing at least!

**Comment by revam** [comment]
@fearnlj01 @ElementalCrisis 
https://github.com/ShokoAnime/ShokoServer/issues/1130

You do know that the groups can contain episodes across types, right? The screenshot @fearnlj01 provided may make the viewer believe that is not possible, but it is. Now if one of you can mange to define the rules you want in plain English then I can turn it into code (if the rules you define aren't too complicated).

**Comment by ElementalCrisis** [comment]
Oh I provided an example but looks like it wasn't included. 

Episodes
Episodes Specials
Specials

If we're tracking type other here as well, since originally, we were not then add that after specials.

**Comment by revam** [comment]
Closed by 81f21870b7dbc03866f876e0c946985487c21db6
